File: Week1/cash.c
#include <stdio.h>
#include <math.h>

int main() {
    float money;
    do {
    printf("Change owed: ");
    scanf("%f", &money);
    } while (money < 0);
    
    money = (int)round(money * 100);
    int count = 0;
    int coins[] = {25, 10, 5, 1};
    while (money > 0) {
        for (int i = 0; i < 4; i++) {
            if (money < coins[i]) {
                continue;
            }
            else {
                count += money / coins[i];
                money -= (money / coins[i]) * coins[i];
            }
        }
    }
    printf("%d\n", count);
    return 0;
}

File: Week6/cash.py
def main():
  change = float(input("Change owed: "))
  count = 0
  coins = [0.25, 0.1, 0.05, 0.01]
  
  for coin in coins:
    change -= change // coin * coin
    count += change // coin
    
  print(int(count))

if __name__ == "__main__":
  main()
